### PR TITLE
Update commit information on CommitDetailsView

### PR DESCRIPTION
**Pull request by sahil143**
## Fixes 

https://issues.redhat.com/browse/HAC-2938

## Description

update commit details view information

## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

image
image

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by MariaLeonova** [comment]
@sahil143 Looks good! I'm wondering if we should make PR a link as well? @Ranelim

File: src/utils/commits-utils.ts [hunk 1]
```diff
@@ -131,3 +133,29 @@ export const showPLRMessage = (plr: PipelineRunKind): string => {
     return 'Releasing';
   }
 };
+
+export const createRepoUrl = (commit: Commit): string | null => {
+  if (commit.repoURL) {
+    return commit.repoURL;
+  }
+  if (commit.repoName && commit.repoOrg) {
+    return `https://github.com/${commit.repoOrg}/${commit.repoName}`;
```
Comment (christianvogt) [inline src/utils/commits-utils.ts:142]: Valid only if `commit.gitProvider === 'github'`

File: src/utils/commits-utils.ts [hunk 2]
```diff
@@ -131,3 +133,29 @@ export const showPLRMessage = (plr: PipelineRunKind): string => {
...
+  }
+  return null;
+};
+
+export const createRepoBranchURL = (commit: Commit): string | null => {
+  const repoUrl = createRepoUrl(commit);
+  if (commit.branch && repoUrl) {
+    return `${repoUrl}/tree/${commit.branch}`;
```
Comment (christianvogt) [inline src/utils/commits-utils.ts:150]: Do we need to limit this to github ?

File: src/utils/commits-utils.ts [hunk 3]
```diff
@@ -131,3 +133,29 @@ export const showPLRMessage = (plr: PipelineRunKind): string => {
...
+  }
+  return null;
+};
+
+export const createRepoPullRequestURL = (commit: Commit): string | null => {
+  const repoURL = createRepoUrl(commit);
+  if (commit.pullRequestNumber && repoURL) {
+    return `${repoURL}/pull/${commit.pullRequestNumber}`;
```
Comment (christianvogt) [inline src/utils/commits-utils.ts:158]: Do we need to limit this to github?

**Review by christianvogt — COMMENTED** [review]
Overall works. Much better title!

File: src/components/Commits/CommitDetailsView.tsx [hunk 4]
```diff
@@ -57,7 +64,7 @@ const CommitDetailsView: React.FC<CommitDetailsViewProps> = ({ commitName, appli
   };
 
   const [isExpanded, setIsExpanded] = React.useState<boolean>(false);
-  const [isFull, setIsFull] = React.useState<boolean>(false);
+  // const [isFull, setIsFull] = React.useState<boolean>(false);
```
Comment (christianvogt) [inline src/components/Commits/CommitDetailsView.tsx:67]: Remove commented out code here and below?

File: src/utils/commits-utils.ts (hunk 1 above)
Comment (sahil143) [reply src/utils/commits-utils.ts:142]: yes, we have checked for git provider wherever this util is used. I will move that check to the methods itself.

File: src/utils/commits-utils.ts (hunk 2 above)
Comment (sahil143) [reply src/utils/commits-utils.ts:150]: moved git provider check inside the util

**Review by rohitkrai03 — APPROVED** [review]
/lgtm

**Pull request merged**